pkgs/tools/graphics/mesa-demos/default.nix

```diff
@@ -25,7 +25,8 @@ stdenv.mkDerivation rec {
     freeglut glew libX11 libXext libGL libGLU mesa wayland
     wayland-protocols
   ] ++ lib.optional (mesa ? osmesa) mesa.osmesa ;
-  nativeBuildInputs = [ meson ninja pkg-config ];
+  nativeBuildInputs = [ meson ninja pkg-config wayland  ];
```

Comment:
```suggestion
  nativeBuildInputs = [ meson ninja pkg-config wayland ];
```

Why wayland as native?

Reply:
Because the build system builds some tooling that links with it, see the original error I posted.

Reply:
Indeed the `wayland` package includes `wayland-scanner` (executable) in addition to `wayland-client` and `wayland-egl` (libs).

Reply:
Does it have split outputs? Can you just depend on `wayland.bin`?

Reply:
Shouldn't nativeBuildInputs do that by default?